src/components/DatePicker/DatePicker.jsx

```diff
@@ -158,10 +158,11 @@ export const DatePicker = ({
     };
 
     // TODO: Should be outside this component because this returns JSX
-    const CaptionElement =
-        shouldShowYearPicker && currentMonth
+    const CaptionElement = useMemo(() => {
+        return shouldShowYearPicker && currentMonth
             ? ({ date }) => <MonthYearSelector date={date} currentMonth={currentMonth} onChange={handleMonthChange} />
             : undefined;
+    }, [shouldShowYearPicker, currentMonth]);
```

Comment:
@VitalyDevico please add dependencies or suppress the warning

Reply:
ah yes, will add suppress now, along with comment, there shouldn't be `handleMonthChange` in depts